### PR TITLE
Refine usage of late disassembler and emitter unit tests

### PR DESCRIPTION
**Pull request by BruceForstall**
Introduce `DOTNET_JitEmitUnitTests`. Set this to the function or functions into which you want the unit tests to be written. E.g., `DOTNET_JitEmitUnitTests=Main` or `DOTNET_JitEmitUnitTests=*`.

Rename `DOTNET_JitDumpEmitUnitTests` to `DOTNET_JitEmitUnitTestsSections`.

Make late disassembler work for altjit: use the "RW" address for the generated code.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@kunalspathak PTAL
cc @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
cc @a74nh

File: src/coreclr/jit/codegenlinear.cpp [hunk 1]
```diff
@@ -2672,12 +2672,12 @@ void CodeGen::genCodeForSetcc(GenTreeCC* setcc)
...
 
 /*****************************************************************************
- * Unit testing of the emitter: If JitDumpEmitUnitTests is set, generate
- * a bunch of instructions, then:
- * Use DOTNET_JitLateDisasm=* to see if the late disassembler thinks the instructions are the same as we do.
- * Or, use DOTNET_JitRawHexCode and DOTNET_JitRawHexCodeFile and disassemble the output file.
+ * Unit testing of the emitter: If JitEmitUnitTests is set for this function, generate
+ * a bunch of instructions, then either:
+ * 1. Use DOTNET_JitLateDisasm=* to see if the late disassembler thinks the instructions are the same as we do. Or,
+ * 2. Use DOTNET_JitRawHexCode and DOTNET_JitRawHexCodeFile and disassemble the output file.
```
Comment (kunalspathak) [inline src/coreclr/jit/codegenlinear.cpp:2678]: nit: just to explain that this is something that happens automatically.

```suggestion
 * 2. Use DOTNET_JitRawHexCode and DOTNET_JitRawHexCodeFile and disassemble the output file with an external disassembler.
```

**Review by kunalspathak — APPROVED** [review]
> Make late disassembler work for altjit: use the "RW" address for the generated code.

Not sure if I followed this change. Why did we have to use `RW` address to make altjit work?

**Comment by BruceForstall** [comment]
> Not sure if I followed this change. Why did we have to use RW address to make altjit work?

Yeah, that's odd. Maybe I was conflating multiple problems. But disassembly seemed to work without using RW addresses, then when I tried altjit it didn't work until I realized other nearby cases were using the RW addresses, and switching to them worked.

**Pull request merged**